plugins/fileservices/fileservices.cpp

```diff
@@ -523,31 +523,63 @@ FILESERVICES_API void FILESERVICES_CALL fsSetReadOnly(ICodeContext *ctx, const c
     throw error.getClear();
 }
 
-
-FILESERVICES_API void FILESERVICES_CALL fsRenameLogicalFile(ICodeContext *ctx, const char *oldname, const char *newname)
+FILESERVICES_API void FILESERVICES_CALL implementRenameLogicalFile(ICodeContext *ctx, const char *oldname, const char *newname, const bool overwrite)
 {
     StringBuffer lfn, nlfn;
     constructLogicalName(ctx, oldname, lfn);
     constructLogicalName(ctx, newname, nlfn);
 
     IDistributedFileTransaction *transaction = ctx->querySuperFileTransaction();
     Linked<IUserDescriptor> udesc = ctx->queryUserDescriptor();
+
+    IDistributedFileDirectory &distributedDirectory = queryDistributedFileDirectory();
+    if (overwrite)
+    {
+        if ( distributedDirectory.exists(nlfn.str(), ctx->queryUserDescriptor(), false, false) &&
+             distributedDirectory.exists(lfn.str(), ctx->queryUserDescriptor(), false, false))
```

Comment:
If old lfn doesn't exist, there's no point in continuing at all, with or without overwrite. (renamePhysical will fail).
So would be clearer if checked if old existed and error out before anything else.

Reply:
Changed